addons/binding/org.openhab.binding.oceanic/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,171 @@
+		<description>Oceanic Water Softener (www.oceanic.be)</description>
+
+		<channels>
+			<channel id="serial" typeId="serial" />
+			<channel id="mac" typeId="mac" />
+			<channel id="name" typeId="name" />
+			<channel id="type" typeId="type" />
+			<channel id="alarm" typeId="alarm" />
+			<channel id="alert" typeId="alert" />
+			<channel id="totalflow" typeId="flow" />
+			<channel id="reserve" typeId="reserve" />
+			<channel id="cycle" typeId="cycle" />
+			<channel id="endofcycle" typeId="time" />
+			<channel id="endofregeneration" typeId="time" />
+			<channel id="hardnessunit" typeId="unit" />
+			<channel id="inlethardness" typeId="hardness" />
+			<channel id="outlethardness" typeId="hardness" />
+			<channel id="cylinderstate" typeId="cylinderstate" />
+			<channel id="salt" typeId="salt" />
+			<channel id="regeneratenow" typeId="regeneratenow" />
+			<channel id="regeneratelater" typeId="regeneratelater" />
+			<channel id="multiregenerate" typeId="multiregenerate" />
+			<channel id="consumptionmonday" typeId="consumption" />
+			<channel id="consumptiontuesday" typeId="consumption" />
+			<channel id="consumptionwednesday" typeId="consumption" />
+			<channel id="consumptionthursday" typeId="consumption" />
+			<channel id="consumptionfriday" typeId="consumption" />
+			<channel id="consumptionsaturday" typeId="consumption" />
+			<channel id="consumptionsunday" typeId="consumption" />
+			<channel id="consumptiontoday" typeId="consumption" />
+			<channel id="consumptionyesterday" typeId="consumption" />
+			<channel id="consumptioncurrentweek" typeId="consumption" />
+			<channel id="consumptionlastweek" typeId="consumption" />
+			<channel id="consumptioncurrentmonth" typeId="consumption" />
+			<channel id="consumptionlastmonth" typeId="consumption" />
+			<channel id="consumptioncomplete" typeId="consumption" />
+			<channel id="consumptionuntreated" typeId="consumption" />
+			<channel id="consumptionpeaklevel" typeId="consumption" />
+			<channel id="pressure" typeId="pressure" />
+			<channel id="maxpressure" typeId="pressure" />
+			<channel id="minpressure" typeId="pressure" />
+			<channel id="maxflow" typeId="flow" />
+			<channel id="lastgeneration" typeId="time" />
+			<channel id="normalregenerations" typeId="number" />
+			<channel id="serviceregenerations" typeId="number" />
+			<channel id="incompleteregenerations" typeId="number" />
+			<channel id="allregenerations" typeId="number" />
+		</channels>
+
+		<config-description>
+			<parameter name="port" type="text" required="true">
+				<label>Serial Port</label>
+				<description>Serial Port of the openHAB host the Oceanic Water Softener is connected to</description>
+			</parameter>
+			<parameter name="interval" type="decimal" required="true">
+				<label>Polling Interval</label>
+				<description>Interval to poll the Oceanic Water Softener for status updates</description>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+
```

Comment:
Could you mark some of these channels as advanced? Or does it really make sense, to have all of them enabled by default?